chess/position_test.go

```diff
@@ -0,0 +1,73 @@
+package chess
+
+import (
+	"math/rand"
+	"reflect"
+	"testing"
+	"testing/quick"
+)
+
+// Generate implements testing/quick.Generator for CastleRights.
+func (CastleRights) Generate(r *rand.Rand, size int) reflect.Value {
+	return reflect.ValueOf(CastleRights(r.Intn(16)))
+}
+
+func TestCastleRights(t *testing.T) {
+	setThenGet := func(a, b CastleRights) bool {
+		a.Set(b)
+		return a.Get(b) || b == 0
+	}
+
+	if err := quick.Check(setThenGet, nil); err != nil {
+		t.Error(err)
+	}
+
+	clearThenGet := func(a, b CastleRights) bool {
+		a.Clear(b)
+		return !a.Get(b)
+	}
+
+	if err := quick.Check(clearThenGet, nil); err != nil {
+		t.Error(err)
+	}
+}
+
+func TestPosition_Move(t *testing.T) {
+	p := NewPosition()
+
+	reset := p.Move(Move{From: E2, To: E4})
+	if !reset {
+		t.Error("Position.Move(): no counter reset after 1. e2e4")
+	}
+
+	e4piece, ok := p.Board.Get(E4)
+	if !ok {
+		t.Error("Position.Move(): no piece at e4")
+	}
+	if e4piece.Color != White || e4piece.Type != Pawn {
```

Comment:
Should we have global variables like:

```go
var WhitePawn = Piece{White, Pawn}
```

to simplify expressions like these?

Reply:
I think we should redefine `Piece` as a `uint8`, so that (a) we can have `const WhitePawn` and (b) we can have `e4piece == WhitePawn`.

Reply:
Done.